### PR TITLE
status messages: change text to make it easier to digest

### PR DESCRIPTION
**Pull request by mrnugget**
This removes the "Your repositories" from the status messages, which was only ever supposed to show up for user-added repositories on dot-com, but has also been showing up for others.

It also updates the links and messages to be more self-explanatory.

## Before & After

| before | after |
|--------| ------|
| before |  after |

## Test plan

- Storybook & existing tests

### CODE REVIEW

File: client/web/src/nav/StatusMessagesNavItem.mocks.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export const allStatusMessages: StatusMessagesResult['statusMessages'] = [
     {
         __typename: 'ExternalServiceSyncError',
         message:
-            'This is a\nmulti line error message\nthat spans multiple lines and also is a bit long on some lines\nbut lets see',
+            'fetching from code host GitHub PRODUCTION: Error getting GitHub repository: nameWithOwner=moby/moby: request to http://github-proxy/repos/moby/moby returned status 401: Bad credentials',
```
Comment (eseliger) [inline client/web/src/nav/StatusMessagesNavItem.mocks.ts:13]: should we also cover a case where there are multiline messages present?

File: client/web/src/nav/StatusMessagesNavItem.tsx [hunk 2]
```diff
@@ -196,7 +196,8 @@ export const StatusMessagesNavItem: React.FunctionComponent<React.PropsWithChild
             return (
                 <StatusMessagesNavItemEntry
                     key="up-to-date"
-                    message="Repositories available for search"
+                    title="Repositories up-to-date"
+                    message="Repositories syncing from from code host and available for search."
```
Comment (eseliger) [inline client/web/src/nav/StatusMessagesNavItem.tsx:200]:
```suggestion
                    message="Repositories syncing from the code host and are available for search."
```

File: client/web/src/nav/StatusMessagesNavItem.tsx [hunk 3]
```diff
@@ -213,9 +214,10 @@ export const StatusMessagesNavItem: React.FunctionComponent<React.PropsWithChild
                             <StatusMessagesNavItemEntry
                                 key={status.message}
                                 message={status.message}
-                                messageHint="Your repositories may not be up-to-date."
+                                title="Cloning repositories"
+                                messageHint="Repository contents may not be up-to-date."
```
Comment (eseliger) [inline client/web/src/nav/StatusMessagesNavItem.tsx:218]: Isn't cloning only the state when we don't have a repo on disk at all? I thought consecutive fetches aren't mirrored here. In that case, the messageHint is a bit confusing to me. May be missing, but not out of date, no?

File: client/web/src/nav/StatusMessagesNavItem.tsx [hunk 4]
```diff
@@ -239,11 +242,12 @@ export const StatusMessagesNavItem: React.FunctionComponent<React.PropsWithChild
                             <StatusMessagesNavItemEntry
                                 key={status.message}
                                 message={status.message}
-                                messageHint="Your repositories may not be up-to-date."
+                                title="Syncing repositories from code hosts"
+                                messageHint="List of repositories may not be up-to-date."
```
Comment (eseliger) [inline client/web/src/nav/StatusMessagesNavItem.tsx]: I don't know what exactly about it to change but "List of repositories" doesn't really convey to me what "the list" is.

File: client/web/src/nav/StatusMessagesNavItem.tsx [hunk 5]
```diff
@@ -270,12 +274,21 @@ export const StatusMessagesNavItem: React.FunctionComponent<React.PropsWithChild
                         />
                     </Tooltip>
                 )}
+                {loading && (
+                    <Tooltip content="Sorry, we couldn’t fetch notifications!">
```
Comment (eseliger) [inline client/web/src/nav/StatusMessagesNavItem.tsx:278]: An error message for loading state? That's kinda weird, isn't it?

File: client/web/src/nav/StatusMessagesNavItem.mocks.ts (hunk 1 above)
Comment (mrnugget) [reply client/web/src/nav/StatusMessagesNavItem.mocks.ts:13]: This is not displayed anymore, so it doesn't matter.

Comment (eseliger) [reply client/web/src/nav/StatusMessagesNavItem.mocks.ts:13]: can we stop fetching the field altogether then?

File: client/web/src/nav/StatusMessagesNavItem.tsx (hunk 3 above)
Comment (mrnugget) [reply client/web/src/nav/StatusMessagesNavItem.tsx:218]: You're right! Something like "Not all repository contents available for search yet." might be better?

File: client/web/src/nav/StatusMessagesNavItem.tsx (hunk 4 above)
Comment (mrnugget) [reply client/web/src/nav/StatusMessagesNavItem.tsx]: Hehe, yeah. Hard to find proper wording

File: client/web/src/nav/StatusMessagesNavItem.tsx (hunk 5 above)
Comment (mrnugget) [reply client/web/src/nav/StatusMessagesNavItem.tsx:278]: Woops! Leftover. I wanted to put a loading spinner in there

File: client/web/src/nav/StatusMessagesNavItem.tsx (hunk 3 above)
Comment (eseliger) [reply client/web/src/nav/StatusMessagesNavItem.tsx:218]: SGTM!

File: client/web/src/nav/StatusMessagesNavItem.tsx (hunk 4 above)
Comment (mrnugget) [reply client/web/src/nav/StatusMessagesNavItem.tsx]: Just talked to Quinn K on this, let's go with `Repositories synced to Sourcegraph may not be up to date.`

**Review by eseliger — APPROVED** [review]

**Comment by quinnkeast** [comment]
I recommend changing "up-to-date" to "up to date" everywhere in the copy. "Up-to-date" would be used as an adjective phrase, like "up-to-date news," while in these contexts it's an adverb phrase, so "Repository contents may not be up to date" is the correct phrasing.

**Comment by mrnugget** [comment]
> I recommend changing "up-to-date" to "up to date" everywhere in the copy.

Thanks! That did struck me as odd.

**Pull request merged**